drivers/dma/dma_stm32_v3.c

```diff
@@ -0,0 +1,690 @@
+		LISTIFY(DT_INST_PROP(index, dma_channels),                                         \
+			DMA_STM32_IRQ_CONNECT_CHANNEL, (;), index);                                \
+	}
+
+#define DMA_STM32_DEFINE_IRQ_HANDLER(chan, dma)                                                    \
+	static void dma_stm32_irq_##dma##_##chan(const struct device *dev)                         \
+	{                                                                                          \
+		dma_stm32_irq_handler(dev, chan);                                                  \
+	}
+
+#define DMA_STM32_INIT_DEV(index)                                                                  \
+	BUILD_ASSERT(DT_INST_PROP(index, dma_channels) == DT_NUM_IRQS(DT_DRV_INST(index)),         \
+		     "Nb of Channels and IRQ mismatch");                                           \
+                                                                                                   \
+	LISTIFY(DT_INST_PROP(index, dma_channels),                                                 \
+		DMA_STM32_DEFINE_IRQ_HANDLER, (;), index);                                         \
+                                                                                                   \
+	DMA_STM32_IRQ_CONNECT(index);                                                              \
+                                                                                                   \
+	static struct dma_stm32_channel dma_stm32_channels_##index[DT_INST_PROP_OR(                \
+		index, dma_channels, DT_NUM_IRQS(DT_DRV_INST(index)))];                            \
+                                                                                                   \
+	static uint32_t dma_stm32_linked_list_buffer##index                               \
+		[DMA_STM32_NUM_DESCRIPTORS_PER_CHANNEL *                                           \
+		 DT_INST_PROP_OR(index, dma_channels, DT_NUM_IRQS(DT_DRV_INST(index)))] __nocache; \
+                                                                                                   \
+	const struct dma_stm32_config dma_stm32_config_##index = {                                 \
+		COND_CODE_1(DT_NODE_HAS_PROP(__node, clocks),                                      \
+			    (.pclken = { .bus = __bus, .enr = __cenr },),                          \
+			    (/* Nothing if clocks not present */))                                 \
+			.config_irq = dma_stm32_config_irq_##index,                                \
+		.base = (DMA_TypeDef *)DT_INST_REG_ADDR(index),                                    \
+		.max_channels =                                                                    \
+			DT_INST_PROP_OR(index, dma_channels, DT_NUM_IRQS(DT_DRV_INST(index))),     \
+		.channels = dma_stm32_channels_##index,                                            \
+		.linked_list_buffer = dma_stm32_linked_list_buffer##index};                        \
```

Comment:
For consistency, prefer:
```suggestion
		.linked_list_buffer = dma_stm32_linked_list_buffer##index,                         \
	};                                                                                         \

Reply:
Done